src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  useContext,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../TodosContext';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    handleCheckboxClick,
+    handleDelete,
+    handleTitleEditing,
+    handleKeyDown,
+    handleBlur,
+  } = useContext(TodosContext);
+  const [editing, setEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(todo.title);
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleField.current && editing) {
+      titleField.current.focus();
+    }
+  }, [editing]);
+
+  return (
+    <li
+      className={cn({
+        completed: todo.completed === true,
+        editing,
+      })}
+    >
+      <div className="view">
+        <input
+          type="checkbox"
+          className="toggle"
+          checked={todo.completed}
+          onClick={() => handleCheckboxClick(todo.id)}
+        />
+        <label
+          htmlFor="toggle-view"
+          onDoubleClick={() => setEditing(true)}
+        >
+          {todo.title}
+        </label>
+        <button
+          aria-label="button"
+          type="button"
+          className="destroy"
+          data-cy="deleteTodo"
+          onClick={() => handleDelete(todo.id)}
+        />
+      </div>
+      <input
+        type="text"
+        className="edit"
+        ref={titleField}
+        value={todoTitle}
+        onChange={(e) => handleTitleEditing(setTodoTitle, e.target.value)}
+        onKeyDown={(e) => handleKeyDown(
+          todo.id,
+          e.key,
+          todoTitle,
+          setTodoTitle,
+          setEditing,
+        )}
+        onBlur={() => handleBlur(
```

Comment:
onBlur method should stay in this component.
But inside onblur method you should call some method from context that will do something to update your todos array.

Reply:
and don't forget to update the whole array.
because now your title change doesn't update the local storage data